src/hubverse_transform/hub_config.py

```diff
@@ -46,6 +48,47 @@ def __repr__(self):
     def __str__(self):
         return f"Hubverse config information for {self.hub_name}."
 
+    def get_task_id_values(self, round_name: str = "all") -> dict[str, set]:
+        """
+        Return a dict of hub task ids and values for a specific round.
+
+        Parameters
+        ----------
+        round_name : str, default="all"
+            Which round's tasks to include. If "all", return tasks
+            and values for all rounds.
+
+        Returns
+        -------
+        model_tasks : dict[str, list]
+            A mapping of tasks ids to their possible values, as
+            defined in a hub's tasks.json configuration file.
+
+        Raises
+        ------
+        ValueError
+            If the tasks configuration file doesn't contain the
+            specified round_name.
+        """
+
+        tasks = self.tasks
+        rounds = tasks.get("rounds", [])
+        if round_name != "all":
+            rounds = [r for r in rounds if r.get("round_name") == round_name]
```

Comment:
In your example below with 
```
                "round_id_from_variable": True,
                "round_id": "reference_date",
                "model_tasks": [
                    {
                        "task_ids": {
                            "reference_date": {"required": None, "optional": ["2024-07-13", "2024-07-21"]},
```
will this find any results if I ask for `round_name = '2024-07-13'`?

Reply:
No, not unless there is a round entry in `tasks.json` that is explicitly named `2024-07-13`.  See note below: https://github.com/hubverse-org/hubverse-transform/pull/22#discussion_r1654950197

I do think we should settle on how to do this with the understanding that 95% of our use case is generating a schema that will work across all hub files to prevent parquet errors when access model-output files on S3.

My vote would be to switch from `round_name` to `round_id` for consistency with the R package: https://github.com/hubverse-org/hubData/blob/main/R/create_hub_schema.R#L101

Reply:
- I agree that we should switch from `round_name` to `round_id`
- "with the understanding that 95% of our use case is generating a schema that will work across all hub files to prevent parquet errors when access model-output files on S3."  But we are going to want these schemas to be correct on a per-round basis, right?  It does not seem correct to generically use the same schema for all rounds.

Reply:
Yeah, it's likely that I overlooked the round_id nuance you suggest when reading the corresponding R function.

To clarify my comment re: the use case....the current goal of ascertaining a hub's schema is to ensure we're applying the correct parquet schema when writing model-output files to S3. To increase the likelihood of successful downstream access, regardless of filter/query patterns, we want a schema that will work for every round and task in `tasks.json`. *

Thus, the plan for using a "get schema" function in the context of `hubverse-transform` is never to specify a round id. I realize there's broader applications (i.e., the Python version of `hubData`), but that's out of scope for solving #14 

I shouldn't have copied over a parameter from the R counterpart and then hand-wave it 😬 Apologies for the confusion.

My preference at this time would be to remove `round_id` and `round_name` and end up with a function that grabs all possible values for every task_id and output_type_id in every round. In other words, focus on the use case of correct schemas on S3.

* This isn't foolproof (for example, if a hub adds a new round with an incompatible schema), but it's better than our current state.

Reply:
I support a focus on the use case of correct schemas [on S3 and elsewhere], but need more convincing on what the right answer is for "correct".  Couple of thoughts/questions related to this:

1. I believe that the use of a single schema across all rounds would result in model output files that fail validation checks as performed by hubValidations, which feels "bad". I would expect data files that live on S3 to pass the same validity checks as data files that live on GitHub or my local computer. Expanding on this/turning it around, it seems like we should be able to have a "to_parquet" type of function that produces parquet-formatted files that pass hubverse validations, as a piece of functionality that's conceptually separate from S3 buckets. Then we drop the resulting files in S3 as part of this transform operation.
2. For non-cloud-based parquet files, do we have a general problem where if I'm trying load model outputs from parquet files submitted across multiple rounds with different schemas, the load fails?

Reply:
@elray1 and I chatted about this and concluded that:

- hubverse-transform should generate schemas on a per-round basis:
    - match the model-output file being transformed to its corresponding round (determined by an explicit `round_id` or, when `round_id_from_variable` is true, the array of values in the `round_id` field)
    - generate the schema from the matched round and use it when read/writing the file
  - our expectation is that multiple rounds associated with a single item in the `task.json` rounds array will have identical schemas (for example, [every round_id that matches one of these reference dates](https://github.com/cdcepi/FluSight-forecast-hub/blob/main/hub-config/tasks.json#L13) will have the same schema)
- because hubverse-transform has access to the round_id via the name of the model-output file, we'll remove round_id/round_name as a pramater to the `get_task_id_values` method

Reply:
Can I ask out of curiosity, why the extra effort to focus on a per round schema? The schema is a hub level property and ideally determined across all rounds. It also seems simpler to determine at a hub level too (at least it was in R, no need to extract and match round_ids) and is what the R version of the function does, with an option to override the data type of the most problematic column, `output_type_id`.

Reply:
I was under the impression that the schema was a per-round property because the columns in submission files and the expected/valid data types of those columns will potentially change from round to round.

Reply:
Column data types shouldn't change from round to round as that's what would cause problems when trying to access files as an arrow dataset. This is primarily a risk for the most problematic column, `output_type_id` which is why early on when we had lots of discussions about this we allowed for a hub to override the automatic detection of the `output_type_id` to the safest most future proof data type, i.e. character.

Admittedly this is not discussed in detail in hubDocs and probably should, and the option to override the `output_type_id` column type has not been propagated as a `hubValidations` setting and also should (and perhaps admins should be able to configure this at hub level too). 

The main point is that we want all files to have the same schema so it should be determined at hub level.

Reply:
if hubValidations is in practice validating against a one-schema-per-hub basis (where that schema may change as rounds are added, implying that old files that passed validations may fail with updated schemas), and given that as Becky noted, exact alignment of file schemas from github hubs to cloud hubs is impossible to achieve if we're changing formats, maybe we should forget about all of my comments here and just go ahead with something.

Reply:
OK!  So with this understanding, I'm on board with removing `round_id` and `round_name` arguments for this functionality so that we just retrieve an overall schema.

Reply:
I think that's the best plan. 

Generally many task IDs that are covered by our schema shouldn't change data type in further rounds as that's somewhat fixed by the schema. Custom task IDs however, which are beyond our control, and the `output_type_id` column have the potential to change and this could indeed cause problems downstream. This is mainly a problem for parquet files.

To reduce the chances of this happening/mitigate the effects we:

- should improve the documentation on this, get admins to think about the issue early on and warn them to avoid changes in data types.
- Should propagate the ability to fix the `output_type_id` column to `hubValidations` and consider a property in the schema where hub admins can configure and communicate this setting. This would give admins the ability to future proof their hubs by setting the column to character if they are unsure whether they may start collecting an output type that could affect the schema.
- As a future feature, once we have created functionality to inspect a hub for integrity, we could also add functionality that could repair any data type discrepancies and update files to conform to a changed schema. This could help admins in a situation where all the above fail and a breaking schema change needs to be introduced.

Happy to open issues regarding the above on Mon if it sounds sensible or a discussion post for more... discussion so as not to take over this PR anymore!

Reply:
that all sounds good to me, i say let's just file issues for these things :)

Reply:
Thanks hopping into the thread @annakrystalli--generating a schema for the entire hub definitely makes things more straightforward 😅